### PR TITLE
Normalizing net_pkt hexdump

### PR DESCRIPTION
**Pull request by tbursztyka**

### CODE REVIEW

File: tests/net/ieee802154/fragment/src/main.c [hunk 1]
```diff
@@ -29,7 +29,7 @@
 #define NET_LOG_ENABLED 1
 #include "net_private.h"
 
-#define DEBUG 0
+#define DEBUG 1
```
Comment (andrewboie) [inline tests/net/ieee802154/fragment/src/main.c]: was this intentional?

File: tests/net/rpl/src/main.c [hunk 2]
```diff
@@ -150,7 +150,7 @@ static int tester_send(struct net_if *iface, struct net_pkt *pkt)
 	DBG("pkt %p to be sent len %lu\n", pkt, net_pkt_get_len(pkt));
 
 #if 0
```
Comment (andrewboie) [inline tests/net/rpl/src/main.c]: why do we have an #if 0
make it in terms of some macro, or delete the whole thing

File: subsys/net/ip/l2/ieee802154/ieee802154.c [hunk 3]
```diff
@@ -32,53 +32,13 @@
 
 #if 0
```
Comment (andrewboie) [inline subsys/net/ip/l2/ieee802154/ieee802154.c]: I know it's outside the patch, but having commented-out code like this is not good.
Make this in terms of some macro IMO

**Review by tyrantlizard77 — CHANGES_REQUESTED** [review]
comments inline

File: tests/net/ieee802154/fragment/src/main.c (hunk 1 above)
Comment (tbursztyka) [reply tests/net/ieee802154/fragment/src/main.c]: It was for testing the patch, but that should not be here indeed.

File: tests/net/rpl/src/main.c (hunk 2 above)
Comment (tbursztyka) [reply tests/net/rpl/src/main.c]: I don't know the background why it is like that, I think the #if can be removed here. Will do that in another patch to make sure net_hexdump_frags() prints out things only when CONFIG_NET_DEBUG_RPL is enabled.

File: subsys/net/ip/l2/ieee802154/ieee802154.c (hunk 3 above)
Comment (tbursztyka) [reply subsys/net/ip/l2/ieee802154/ieee802154.c]: A Kconfig option will do yes.

File: tests/net/rpl/src/main.c (hunk 2 above)
Comment (jukkar) [reply tests/net/rpl/src/main.c]: Please do not remove the "#if 0" from this rpl test program. These were there on purpose as some of the tests need more TLC but there has not been time to fix/enhance them.

File: subsys/net/ip/net_private.h [hunk 4]
```diff
@@ -214,14 +223,37 @@ static inline void net_hexdump(const char *str, const u8_t *packet,
...
-/* Hexdump from all fragments */
-static inline void net_hexdump_frags(const char *str, struct net_pkt *pkt)
+static inline void net_hexdump(const char *str,
+			       const u8_t *packet, size_t length)
 {
+	if (!length) {
+		SYS_LOG_DBG("%s zero-length packet", str);
+		return;
+	}
+
```
Comment (jukkar) [inline subsys/net/ip/net_private.h]: Print header (the str variable) here too.

**Review by jukkar — CHANGES_REQUESTED** [review]

File: tests/net/rpl/src/main.c (hunk 2 above)
Comment (tbursztyka) [reply tests/net/rpl/src/main.c]: I am not removing the #if 0 at line 535, there the comment specifically says it should stay and need more work. It's not the same issue.

File: subsys/net/ip/net_private.h (hunk 4 above)
Comment (tbursztyka) [reply subsys/net/ip/net_private.h]: indeed

**Review by jukkar — APPROVED** [review]

**Comment by jukkar** [comment]
Ping @andrewboie are you ok with current version?

**Review by pfalcon — APPROVED** [review]

**Comment by jukkar** [comment]
@andrewboie Ping, we need your approval in order to merge this one.

**Pull request merged**